### PR TITLE
Add Build Run Policy section

### PR DESCRIPTION
**Pull request by mfojtik**
Origin PR: https://github.com/openshift/origin/pull/8453

### CODE REVIEW

**Comment by mfojtik** [comment]
@bparees @adellape PTAL

(this is not merge-able until the origin PR is merged).

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -82,41 +83,44 @@ spec:
...
+  postCommit: <7>
       script: "bundle exec rake test"
 ----
 
 <1> This specification will create a new `*BuildConfig*` named
 *ruby-sample-build*.
-<2> You can specify a list of link:#build-triggers[triggers], which cause a new
+<2> The `*runPolicy*` field controls how the builds created from this build
+configuration should run. The default value is `*Serial*`, which means new builds
+will run in serial order.
```
Comment (bparees) [inline dev_guide/builds.adoc]: slight rephrasing:

The \`\*runPolicy\*\` field controls whether builds created from this build
configuration can be run simultaneously. The default value is `*Serial*`, which means new builds
will run in serially, not simultaneously.

(yeah i know it kinda ignores SerialLatestOnly, but i don't think we need to try to explain that in the footnote here)

File: dev_guide/builds.adoc [hunk 2]
```diff
@@ -1761,6 +1765,130 @@ This example uses `*pullSecret*` in a Source build, but it is also applicable
...
+
+It is also possible to change the `*runPolicy*` value for existing build
+configurations. 
+
+  * Changing `*Parallel*` to `*Serial*` or `*SerialLatestOnly*` and triggering a
+    new build from this configuration will cause the new build to wait until all
+    parallel builds complete as the "serial" build can only run alone.
+  * Changing `*Serial*` to `*SerialLatestOnly*` and triggering a new build will
+    cause cancellation of all existing builds in queue, except the currently
+    running build. The new build will execute as next.
```
Comment (bparees) [inline dev_guide/builds.adoc]: "will cause cancellation of all existing builds in queue, except the currently running build and the most recently created build. The newest build will execute next."

File: dev_guide/builds.adoc [hunk 3]
```diff
@@ -1761,6 +1765,130 @@ This example uses `*pullSecret*` in a Source build, but it is also applicable
...
+
+[[build-serial-run-policy]]
+=== Serial Run Policy
+
+Setting the `*runPolicy*` field to `*Serial*` will cause all new builds created
+from the `*Build*` configuration to be run in "serial" order. That means there
```
Comment (bparees) [inline dev_guide/builds.adoc]: no scare quotes around serial.  we mean serial literally.
https://en.wikipedia.org/wiki/Scare_quotes

File: dev_guide/builds.adoc [hunk 4]
```diff
@@ -1761,6 +1765,130 @@ This example uses `*pullSecret*` in a Source build, but it is also applicable
...
+When the `sample-build-1` build completes, the `sample-build-2` build will run:
+
+====
+----
+NAME             TYPE      FROM          STATUS    STARTED          DURATION
+sample-build-1   Source    Git@e79d887   Completed 43 seconds ago   34s
+sample-build-2   Source    Git@1aa381b   Running   2 seconds ago    2s
+sample-build-3   Source    Git           New
+----
+====
```
Comment (bparees) [inline dev_guide/builds.adoc]: i like the example!

File: dev_guide/builds.adoc [hunk 5]
```diff
@@ -1761,6 +1765,130 @@ This example uses `*pullSecret*` in a Source build, but it is also applicable
...
+
+
+[[build-serial-latest-only-run-policy]]
+=== SerialLatestOnly Run Policy
+
+Setting the `*runPolicy*` field to `*SerialLatestOnly*` will cause all new
+builds created from the `*Build*` configuration to be run in "serial" order,
```
Comment (bparees) [inline dev_guide/builds.adoc]: same here, no scare quotes.

File: dev_guide/builds.adoc [hunk 6]
```diff
@@ -1761,6 +1765,130 @@ This example uses `*pullSecret*` in a Source build, but it is also applicable
...
+same as using the `*Serial*` run policy. The difference is that when a currently
+running build completes, the next build that will run is the latest build
+created. In other words, you don't wait for the queued builds to run as they
+will be skipped. This policy can be used for fast, iterative development.
```
Comment (bparees) [inline dev_guide/builds.adoc]: "will be skipped.  Skipped builds are marked as \`\*Cancelled\*\`."  This policy can be..."

File: dev_guide/builds.adoc [hunk 7]
```diff
@@ -1761,6 +1765,130 @@ This example uses `*pullSecret*` in a Source build, but it is also applicable
...
+
+[[build-parallel-run-policy]]
+=== Parallel Run Policy
+
+Setting the `*runPolicy*` field to `*Parallel*` will cause all new builds
+created from the `*Build*` configuration to be run in parallel.  This can
+produce unpredictable results as the first created build can complete last which
+and will replace the pushed Docker image produced by the last build which
+completed earlier.
+Use the parallel run policy in case when you don't care about the order in which
```
Comment (bparees) [inline dev_guide/builds.adoc]: s/in case/in cases/
s/when/where/

**Comment by bparees** [comment]
couple nits but mostly looks great.  the examples a a great touch.

**Comment by mfojtik** [comment]
@bparees i was thinking about animated gifs originally :-))

**Comment by mfojtik** [comment]
@bparees all fixed, thanks!

File: dev_guide/builds.adoc [hunk 8]
```diff
@@ -82,41 +83,44 @@ spec:
...
+  postCommit: <7>
       script: "bundle exec rake test"
 ----
 
 <1> This specification will create a new `*BuildConfig*` named
 *ruby-sample-build*.
-<2> You can specify a list of link:#build-triggers[triggers], which cause a new
+<2> The `*runPolicy*` field controls whether builds created from this build
+configuration can be run simultaneously. The default value is *Serial*, which means new builds
+will run in serially, not simultaneously.
```
Comment (bparees) [inline dev_guide/builds.adoc]: s/in//

Comment (bparees) [inline dev_guide/builds.adoc]: (you need to grammar check my suggestions! :) )

**Comment by bparees** [comment]
one last nit, sorry.  was an error in my suggestion.

File: dev_guide/builds.adoc (hunk 8 above)
Comment (mfojtik) [inline dev_guide/builds.adoc]: i will also s/serially/sequentially/ :-)

**Comment by bparees** [comment]
lgtm. @openshift/team-documentation ptal, can merge after https://github.com/openshift/origin/pull/8453 goes in.

File: dev_guide/builds.adoc (hunk 4 above)
Comment (jhadvig) [inline dev_guide/builds.adoc]: +1

**Comment by php-coder** [comment]
https://github.com/openshift/origin/pull/8453 has been merged by the way..

File: dev_guide/builds.adoc [hunk 9]
```diff
@@ -71,7 +71,8 @@ apiVersion: "v1"
 metadata:
   name: "ruby-sample-build" <1>
 spec:
-  triggers: <2>
+  runPolicy: Serial <2>
```
Comment (php-coder) [inline dev_guide/builds.adoc]: Should we add quotes to the value?

Comment (ahardin-rh) [inline dev_guide/builds.adoc]: @php-coder @mfojtik yes, to remain consistent with the other areas, quotes should be added to the value.

Also, looks like we'll need a rebase. Thanks!

Comment (ahardin-rh) [inline dev_guide/builds.adoc]: @mfojtik @php-coder bump

Comment (bparees) [inline dev_guide/builds.adoc]: @mfojtik is on PTO until next week i think, maybe re-ping him then.

Comment (ahardin-rh) [inline dev_guide/builds.adoc]: @bparees Thanks!

Comment (bparees) [inline dev_guide/builds.adoc]: @mfojtik bump. needs quotes and a rebase.

**Comment by bparees** [comment]
@openshift/team-documentation @mfojtik is swamped w/ release shutdown work, any chance you guys can cherry-pick and rebase this and fix the nits so we can get it merged?

**Comment by tpoitras** [comment]
Paging Dr. @vikram-redhat :wrench:

**Comment by vikram-redhat** [comment]
@bparees @tpoitras  - looking

**Comment by tpoitras** [comment]
Hey @bparees is this for enterprise 3.2, 3.1, 3.0? Dedicated? Online?

**Comment by bparees** [comment]
3.3, online, origin.

Ben Parees | OpenShift

**Comment by bparees** [comment]
And dedicated.

Ben Parees | OpenShift

**Comment by vikram-redhat** [comment]
@ahardin-rh - saw this is already assigned to you. Can you rebase and fix and merge to master?

**Comment by ahardin-rh** [comment]
@vikram-redhat @bparees I'm on it! :hammer:

**Comment by ahardin-rh** [comment]
Work is now moved to https://github.com/openshift/openshift-docs/pull/2681

**Pull request closed without merging**